src/archive.rs

```diff
@@ -243,19 +246,77 @@ impl<'a> EntriesFields<'a> {
             return Err(other("archive header checksum mismatch"))
         }
 
+        let data_size = try!(header.data_size());
         let size = try!(header.size());
+        let sparse_chunks = if let Some(gnu) = header.as_gnu_mut() {
+            let mut chunks = Vec::new();
+            for item in &gnu.sparse[..] {
+                if item.offset[0] != 0 && item.numbytes[0] != 0 {
+                    let off = try!(octal_from(&item.offset[..]));
+                    let nbytes = try!(octal_from(&item.numbytes[..]));
+                    if nbytes > 0 {
+                        chunks.push((off, nbytes));
+                    }
+                }
+            }
+            if gnu.isextended[0] == 1 {
+                loop {
+                    let mut ext_header = [0u8; 512];
+                    try!(read_all(&mut &self.archive.inner, &mut ext_header));
+                    self.next += 512;
+                    let sparse_header: &GnuExtSparseHeader;
+                    sparse_header = unsafe { transmute(&ext_header) };
```

Comment:
I'd prefer if this had a similar `as_mut_bytes` method like the other header entries to make this a little safer